File: JS101_programming_foundations/small_problems/05_easy_5/ex_10.js
// Write a function that takes one argument, an array of integers, and returns the average of all the integers in the array, rounded down to the integer component of the average.The array will never be empty, and the numbers will always be positive integers.

function average(array) {
  return Math.floor(array.reduce((sum, num) => sum + num) / array.length);
}

average([1, 5, 87, 45, 8, 8]);       // 25
average([9, 47, 23, 95, 16, 52]);    // 40

File: RB101_programming_foundations/exercises/medium_2/ex_10.rb
def sum_square_difference(num)
  sum = 0
  sum_of_squares = 0

  1.upto(num) do |value|
    sum += value
    sum_of_squares = sum_of_squares + (value**2)
  end
  
  sum**2 - sum_of_squares
end



p sum_square_difference(3) == 22
   # -> (1 + 2 + 3)**2 - (1**2 + 2**2 + 3**2)
p sum_square_difference(10) == 2640
p sum_square_difference(1) == 0
p sum_square_difference(100) == 25164150